from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtGui import QStandardItem


class Ui_ViewOrdine(object):
    def __init__(self, ordine):
        self.ordine = ordine

    def setupUi(self, ViewOrdine):
        ViewOrdine.setObjectName("ViewOrdine")
        ViewOrdine.resize(673, 659)
        self.listView = QtWidgets.QListView(ViewOrdine)
        self.listView.setGeometry(QtCore.QRect(50, 70, 571, 371))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Text, brush)
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Text, brush)
        brush = QtGui.QBrush(QtGui.QColor(120, 120, 120))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Text, brush)
        self.listView.setPalette(palette)
        font = QtGui.QFont()
        font.setFamily("Harlow Solid Italic")
        font.setPointSize(16)
        font.setItalic(True)
        self.listView.setFont(font)
        self.listView.setObjectName("listView")
        self.Carrello = QtWidgets.QLabel(ViewOrdine)
        self.Carrello.setGeometry(QtCore.QRect(160, 10, 401, 51))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(120, 120, 120))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        self.Carrello.setPalette(palette)
        font = QtGui.QFont()
        font.setFamily("Harlow Solid Italic")
        font.setPointSize(26)
        font.setItalic(True)
        self.Carrello.setFont(font)
        self.Carrello.setAccessibleName("")
        self.Carrello.setObjectName("Carrello")
        self.pushButton_3 = QtWidgets.QPushButton(ViewOrdine)
        self.pushButton_3.setGeometry(QtCore.QRect(170, 590, 291, 61))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.ButtonText, brush)
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.ButtonText, brush)
        brush = QtGui.QBrush(QtGui.QColor(120, 120, 120))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.ButtonText, brush)
        self.pushButton_3.setPalette(palette)
        font = QtGui.QFont()
        font.setFamily("Harlow Solid Italic")
        font.setPointSize(24)
        font.setItalic(True)
        self.pushButton_3.setFont(font)
        self.pushButton_3.setObjectName("pushButton_3")
        self.Carrello_2 = QtWidgets.QLabel(ViewOrdine)
        self.Carrello_2.setGeometry(QtCore.QRect(0, 510, 111, 51))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(120, 120, 120))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        self.Carrello_2.setPalette(palette)
        font = QtGui.QFont()
        font.setFamily("Harlow Solid Italic")
        font.setPointSize(20)
        font.setItalic(True)
        self.Carrello_2.setFont(font)
        self.Carrello_2.setAccessibleName("")
        self.Carrello_2.setObjectName("Carrello_2")
        self.Carrello_3 = QtWidgets.QLabel(ViewOrdine)
        self.Carrello_3.setGeometry(QtCore.QRect(10, 550, 651, 51))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(120, 120, 120))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        self.Carrello_3.setPalette(palette)
        font = QtGui.QFont()
        font.setFamily("Harlow Solid Italic")
        font.setPointSize(18)
        font.setItalic(True)
        self.Carrello_3.setFont(font)
        self.Carrello_3.setAccessibleName("")
        self.Carrello_3.setObjectName("Carrello_3")
        self.Carrello_4 = QtWidgets.QLabel(ViewOrdine)
        self.Carrello_4.setGeometry(QtCore.QRect(210, 440, 111, 51))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(120, 120, 120))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        self.Carrello_4.setPalette(palette)
        font = QtGui.QFont()
        font.setFamily("Harlow Solid Italic")
        font.setPointSize(20)
        font.setItalic(True)
        self.Carrello_4.setFont(font)
        self.Carrello_4.setAccessibleName("")
        self.Carrello_4.setObjectName("Carrello_4")
        self.Carrello_5 = QtWidgets.QLabel(ViewOrdine)
        self.Carrello_5.setGeometry(QtCore.QRect(320, 440, 111, 51))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(206, 186, 115))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(120, 120, 120))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        self.Carrello_5.setPalette(palette)
        font = QtGui.QFont()
        font.setFamily("Harlow Solid Italic")
        font.setPointSize(20)
        font.setItalic(True)
        self.Carrello_5.setFont(font)
        self.Carrello_5.setAccessibleName("")
        self.Carrello_5.setObjectName("Carrello_5")

        self.model = QtGui.QStandardItemModel()
        self.listView.setModel(self.model)

        self.retranslateUi(ViewOrdine)
        QtCore.QMetaObject.connectSlotsByName(ViewOrdine)

    def retranslateUi(self, ViewOrdine):
        _translate = QtCore.QCoreApplication.translate
        ViewOrdine.setWindowTitle(_translate("ViewOrdine", "Ordine"))
        self.Carrello.setText(_translate("ViewOrdine", "Prodotti dell\'ordine"))
        self.pushButton_3.setText(_translate("ViewOrdine", "Elimina Ordine"))
        self.Carrello_2.setText(_translate("ViewOrdine", "Codice: "))
        self.Carrello_3.setText(_translate("ViewOrdine", self.ordine.codice))
        self.Carrello_4.setText(_translate("ViewOrdine", "Totale:"))
        app = str(self.ordine.get_totale())
        stringa = app + "€"
        self.Carrello_5.setText(_translate("ViewOrdine", stringa))


        for i in  self.ordine.lista_prodotti:
            app1 = i.get_nome()
            app2 = i.personalizzazione.tipo

            app3 = float(i.prezzo)
            app4 = float(i.personalizzazione.prezzo)
            prezzo = app3 + app4
            app5 = str(prezzo)
            stri = app1 + " ( " + app2 + ")    " + app5 + "€"
            item = QStandardItem()
            item.setText(stri)
            item.setEditable(False)
            font = item.font()
            font.setPointSize(18)
            item.setFont(font)
            self.model.appendRow(item)




if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    ViewOrdine = QtWidgets.QDialog()
    ui = Ui_ViewOrdine()
    ui.setupUi(ViewOrdine)
    ViewOrdine.show()
    sys.exit(app.exec_())
